### PR TITLE
[New Model]: Mandatory Dismantling

### LINKED ISSUE #493: [New Model]: Mandatory Dismantling
## Model Description

With the Mandatory Dismantling data model, we want to mark the components in the vehicle with the notice for mandatory dismantling. 

## Relevant Standards

ANHÄNGE des Vorschlags für eine VERORDNUNG DES EUROPÄISCHEN PARLAMENTS UND DES RATES
über Anforderungen an die kreislauforientierte Konstruktion von Fahrzeugen und über 
die Entsorgung von Altfahrzeugen, zur Änderung der Verordnungen (EU) 2018/858 und 
(EU) 2019/1020 und zur Aufhebung der Richtlinien 2000/53/EG und 2005/64/EG

## Example Data

Screenshot 
Mandatory Dismantling

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by tanweersalah**
## Description

With the Mandatory Dismantling data model, we want to mark the components in the vehicle with the notice for mandatory dismantling. This PR represents the Aspect Model "MandatoryDismantling".

Issue - https://github.com/eclipse-tractusx/sldt-semantic-models/issues/493
 -->

Closes #493 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.5.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by agg3fe** [comment]
please remove gen folder from files added to PR. @tanweersalah

**Comment by tanweersalah** [comment]
removed @agg3fe

**Comment by agg3fe** [comment]
@tanweersalah is this PR ready for review?

**Comment by tanweersalah** [comment]
@agg3fe , we are working on a few changes, will update you once it is ready

**Comment by tanweersalah** [comment]
@agg3fe  please review, Thanks

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 1]
```diff
@@ -0,0 +1,994 @@
+#######################################################################
+# Copyright (c) 2023, 2024 SAP SE
```
Comment (agg3fe) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:2]: only 2024 required

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 2]
```diff
@@ -0,0 +1,994 @@
...
+   samm:preferredName "Fluid"@en ;
+   samm:description "Information on the different fluid and liquid available in the asset"@en ;
+   samm:characteristic :FluidCharacteristic .
+
+:component a samm:Property ;
+   samm:preferredName "Component"@en ;
+   samm:description "information on the different components available in the asset"@en ;
+   samm:characteristic :ComponentCharacteristic .
+
+:time a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:189]: Property and the referenced Characteristic should not have the same name

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 3]
```diff
@@ -0,0 +1,994 @@
...
+   samm:preferredName "Time"@en ;
+   samm:description "Time of the specific depollution."@en ;
+   samm:characteristic :Time ;
+   samm:exampleValue "14:34:00"^^xsd:time .
+
+:date a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:195]: Property and the referenced Characteristic should not have the same name

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 4]
```diff
@@ -0,0 +1,994 @@
...
+   samm:properties ( :weight :requiredTool :permanentMagnet ) .
+
+:itemType a samm:Property ;
+   samm:preferredName "itemType"@en ;
+   samm:description "The item type of parent."@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "Diesel" .
+
+:itemDetail a samm:Property ;
+   samm:preferredName "ItemDetail"@en ;
```
Comment (agg3fe) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:947]: preferredName should be human readable and follow normal orthography (e.g., no camel case but normal word separation).

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 5]
```diff
@@ -0,0 +1,994 @@
...
+   samm:description "Encapsulates the Battery detail."@en ;
+   samm:properties ( :batteryType :dischargeInstruction :requiredTool :weight ) .
+
+:MotorDetailEntity a samm:Entity ;
+   samm:preferredName "Motor Detail Entity"@en ;
+   samm:description "Encapsulates the Motor detail."@en ;
+   samm:properties ( :weight :requiredTool :permanentMagnet ) .
+
+:itemType a samm:Property ;
+   samm:preferredName "itemType"@en ;
```
Comment (agg3fe) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:941]: preferredName should be human readable and follow normal orthography (e.g., no camel case but normal word separation).

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 6]
```diff
@@ -0,0 +1,994 @@
...
+   samm:preferredName "Transmission Component"@en ;
+   samm:description "A property of the model, named transmission component."@en ;
+   samm:characteristic :TransmissionComponentCharacteristic .
+
+:QuantityCharacteristic a samm-c:Quantifiable ;
+   samm:preferredName "Quantity Characteristic"@en ;
+   samm:description "Describes a Property which contains a number."@en ;
+   samm:dataType xsd:integer .
+
+:WeightCharacteristic a samm-c:Measurement ;
```
Comment (catroest) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:358]: I would suggest you look into the shared quantity model. It describes for example weight values, (where you can also choose the unit)

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check all the comments

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl (hunk 4 above)
Comment (tanweersalah) [reply io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl]: Fixed

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl (hunk 3 above)
Comment (tanweersalah) [reply io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:196]: Fixed

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl (hunk 2 above)
Comment (tanweersalah) [reply io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:190]: Fixed

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl (hunk 1 above)
Comment (tanweersalah) [reply io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl]: Updated

**Review by agg3fe — APPROVED** [review]
LGTM

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 7]
```diff
@@ -0,0 +1,991 @@
...
+:ComponentCharacteristic a samm:Characteristic ;
+   samm:preferredName "Component Characteristic"@en ;
+   samm:description "The characteristics of the Component."@en ;
+   samm:dataType :ComponentEntity .
+
+:TimeCharacteristic a samm-c:Quantifiable ;
+   samm:preferredName "Time Characteristic"@en ;
+   samm:dataType xsd:time .
+
+:DateCharacteristic a samm-c:Quantifiable ;
```
Comment (catroest) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:379]: missing description

Comment (catroest) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:379]: missing description

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 8]
```diff
@@ -0,0 +1,991 @@
...
+   samm:preferredName "Date Characteristic"@en ;
+   samm:dataType xsd:date .
+
+:Number a samm-c:Quantifiable ;
```
Comment (catroest) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:383]: missing preffered name and description

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 9]
```diff
@@ -0,0 +1,991 @@
...
+   samm:preferredName "Fluid Characteristic"@en ;
+   samm:description "The characteristics of the Fluid."@en ;
+   samm:dataType :FluidEntity .
+
+:ComponentCharacteristic a samm:Characteristic ;
+   samm:preferredName "Component Characteristic"@en ;
+   samm:description "The characteristics of the Component."@en ;
+   samm:dataType :ComponentEntity .
+
+:TimeCharacteristic a samm-c:Quantifiable ;
```
Comment (catroest) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:375]: missing description

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl [hunk 10]
```diff
@@ -0,0 +1,991 @@
...
+   samm:characteristic ext-uuid:UuidV4Trait ;
+   samm:exampleValue "urn:uuid:580d3adf-1981-44a0-a214-13d6ceed9379" .
+
+:van a samm:Property ;
+   samm:preferredName "VAN"@en ;
+   samm:description "Vehicle Anonymiyed Number. This should be deducted from a reference to the vehicle TWIN"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "WBA41DU060S228332" .
+
+:bbpNumberDismantler a samm:Property ;
```
Comment (catroest) [inline io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:53]: is this a typo and you mean bpNumberDismanter = bpnDIsmanter? What is bbp?

**Review by catroest — COMMENTED** [review]
Some missing descriptions and one clarification.
One general remark: you could combine date and time in one property and use the characterisitc with the type DateTime. (If you do not have reasons to explicitly separate them)

**Comment by agg3fe** [comment]
@tanweersalah could you please check the above comments.

File: io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl (hunk 10 above)
Comment (janinemuegge) [reply io.catenax.mandatory_dismantling/1.0.0/MandatoryDismantling.ttl:53]: Yes, that's a typo. bpNumber Dismantler mean bpnDismantler

File: io.catenax.mandatory_dismantling/RELEASE_NOTES.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+All notable changes to this model will be documented in this file.
+
+## [1.0.0] - 2024-03-04
```
Comment (agg3fe) [inline io.catenax.mandatory_dismantling/RELEASE_NOTES.md:4]: please change the date to 11th march.

**Pull request merged**